### DISCUSSION TITLE
Using custom fields in the user table and how to use them from auth()->user()

### DISCUSSION THREAD

**Discussion by yuyake** [codeigniter4/shield#846] [Q&A] [answered]
"I am a beginner, and thanks to CodeIgniter Shield, I am developing easily.

I have added user columns to the user table and created a model that inherits from ShieldUserModel. The registration and login functions are working well, as follows:

php
Copy code
```php
<?php

declare(strict_types=1);

namespace App\Models;

use CodeIgniter\Shield\Models\UserModel as ShieldUserModel;

class UserModel extends ShieldUserModel
{
    protected function initialize(): void
    {
        parent::initialize();

        // Processing when fields are added to the users table
        $this->allowedFields = [
            ...$this->allowedFields,
            'phone_number',
            'authname',
        ];
    }
}
```

What I want to achieve is that when I call `$user = auth()->user();` as in the manual, the `$user` object should contain the contents defined in `UserIdentityModel` implemented in Shield. I want to be able to call phone_number, authname, etc., in the same way as `$user->phone_number;`.

From what I have found on forums and Google searches, is extending Shield Entities the only way to do this? Do you have a simple example?"

**Comment by kenjis** [comment, marked as answer]
Can't you use `$user->phone_number` now?

**Reply by yuyake** [reply]
@kenjis 
I'm glad to hear that your issue has been resolved by working with the UserModel. It sounds like you initially encountered some challenges, including red lines in PHPStorm due to IDE errors and then encountering further errors when working with the Entity feature. However, after reviewing my response, you were able to confirm that using the initial UserModel resolved the issue without any problems.

**Reply by kenjis** [reply]
Sorry, do you mean like the following? 

> I'm glad that my issue has been resolved by working with the UserModel. It seems like I initially encountered some challenges, including red lines in PHPStorm due to IDE errors and then encountering further errors when working with the Entity feature. However, after reviewing your response, I was able to confirm that using the initial UserModel resolved the issue without any problems.
